add playlist progress bar

makerio90 opened the pull request:
could use some css

---
add playlist progress bar
---

**Important note**
We may remove your pull request if you do not use this provided PR template correctly.

**Pull Request Type**
Please select what type of pull request this is:
- [ ] Bugfix
- [x] Feature Implementation

**Related issue**
Please link the issue your pull request is referring to. If this pull request fully resolves the relevant issue, put "closes" before the issue number. Example: "closes #123456".

**Description**
when whatching a playlist, it displays a little progress bar to show you your progress through the playlist.

**Screenshots (if appropriate)**
Please add before and after screenshots if there is a visible change.
![image](https://user-images.githubusercontent.com/78005038/179372152-c65342f8-bf5f-475c-966c-02841d235250.png)

**Testing (for code that is not small enough to be easily understandable)**
Has this pull request been tested? y
Please describe shortly how you tested it and whether there are any ramifications remaining. 
it works.
**Desktop (please complete the following information):**
 - OS: [manjaro]
 - OS Version: [n/a]
 - FreeTube version: [latest]

**Additional context**
Add any other context about the problem here.
needs css

efb4f5ff-1298-471a-8973-3d47447115dc reviewed (changes requested):
fix lint warnings

Would also like to see the color of the progress bar gets adjusted to the color u choose in the settings under Theme settings > secondary theme settings

makerio90 commented:
> fix lint warnings
> 
> Would also like to see the color of the progress bar gets adjusted to the color u choose in the settings under Theme settings > secondary theme settings

fixed and fixed, although the theming is a bit odd.

efb4f5ff-1298-471a-8973-3d47447115dc commented:
30k lines added? I think something didnt go well.

makerio90 commented:
> 30k lines added and 7k removed? I think something didnt go well.

i dont think i was supposed to track yarn.lock or package-lock.json

efb4f5ff-1298-471a-8973-3d47447115dc reviewed (approved):
LGTM

PikachuEXE commented on src/renderer/components/watch-video-playlist/watch-video-playlist.vue:
```diff
@@ -22,6 +22,11 @@
       >
         - {{ currentVideoIndex }} / {{ playlistVideoCount }}
+        <progress
+          class="playlistProggressBar"
```
```suggestion
          class="playlistProgressBar"
```

PrestonN commented:
Not too sure if having a progress bar will make sense 100% of the time. What if the user is shuffling through the playlist? This doesn't seem to account for reverse order of playlists either. I don't see this as being as beneficial as we might think it will be. Open to other opinions however.

PikachuEXE reviewed (changes requested):
What about only showing this when the order is "ascending" (with shuffling disabled)
We can figure out if that's useful for "reverse order" (I don't even know that exists)

But yup in this case this new components should only be displayed sometimes

makerio90 commented:
> What about only showing this when the order is "ascending" (with shuffling disabled) We can figure out if that's useful for "reverse order" (I don't even know that exists)
> 
> But yup in this case this new components should only be displayed sometimes
fixed
Im checking for `!shuffleEnabled && !reversePlaylist`

PikachuEXE reviewed (approved):
Tested locally

https://user-images.githubusercontent.com/1018543/180117295-0bb8b1c6-7ba1-456e-89aa-c6f4d9899bc0.mov

efb4f5ff-1298-471a-8973-3d47447115dc reviewed (approved):
LGTM

efb4f5ff-1298-471a-8973-3d47447115dc commented:
@absidue or @ChunkyProgrammer one more approval needed. Review if u got some spare time.

absidue commented:
When the page is between 900px and 1292px wide the progress bar will wrap around to a new line if the channel name or current index are "too long", which looks bad, ideally the progress bar would stay on the same line as the channel name and the numbers, just changing it's width to fit into the available space.
![playlist-progress-bar](https://user-images.githubusercontent.com/48293849/184853612-fc942729-d8be-4818-bfed-c90efcfb5004.jpg)

absidue commented:
Okay so I realised that fixing that issue requires more CSS and changing the layout, so I'm happy getting this merged and I'll create a PR to fix that issue afterwards. That way I can also make sure that the CSS changes I want to make will work with the newer CSS on the development branch (I tried rebasing but the commits with the package.json, package-lock.json and yarn.lock files break rebasing).
